exporter/clickhouseexporter/internal/histogram_metrics.go

```diff
@@ -0,0 +1,179 @@
+    MetricUnit,
+    Attributes,
+	StartTimeUnix,
+	TimeUnix,
+	Count,
+	Sum,
+	BucketCounts,
+	ExplicitBounds,
+  	Exemplars.FilteredAttributes,
+	Exemplars.TimeUnix,
+    Exemplars.ValueAsDouble,
+    Exemplars.ValueAsInt,
+    Exemplars.SpanId,
+    Exemplars.TraceId,
+	Flags,
+	Min,
+	Max) VALUES `
+	histogramPlaceholders = "(?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)"
+)
+
+type histogramModel struct {
+	metricName        string
+	metricDescription string
+	metricUnit        string
+	histogram         pmetric.Histogram
+}
+
+type HistogramMetrics struct {
+	histogramModel []*histogramModel
+	metadata       *MetricsMetaData
+	insertSQL      string
+}
+
+func (h *HistogramMetrics) insert(ctx context.Context, tx *sql.Tx, logger *zap.Logger) error {
+	var valuePlaceholders []string
+	var valueArgs []interface{}
+
+	for _, model := range h.histogramModel {
+		for i := 0; i < model.histogram.DataPoints().Len(); i++ {
+			dp := model.histogram.DataPoints().At(i)
+			valuePlaceholders = append(valuePlaceholders, histogramPlaceholders)
+
+			valueArgs = append(valueArgs, h.metadata.ResAttr)
+			valueArgs = append(valueArgs, h.metadata.ResURL)
+			valueArgs = append(valueArgs, h.metadata.ScopeInstr.Name())
+			valueArgs = append(valueArgs, h.metadata.ScopeInstr.Version())
+			valueArgs = append(valueArgs, attributesToMap(h.metadata.ScopeInstr.Attributes()))
+			valueArgs = append(valueArgs, h.metadata.ScopeInstr.DroppedAttributesCount())
+			valueArgs = append(valueArgs, h.metadata.ScopeURL)
+			valueArgs = append(valueArgs, model.metricName)
+			valueArgs = append(valueArgs, model.metricDescription)
+			valueArgs = append(valueArgs, model.metricUnit)
+			valueArgs = append(valueArgs, attributesToMap(dp.Attributes()))
+			valueArgs = append(valueArgs, dp.StartTimestamp().AsTime().UnixNano())
+			valueArgs = append(valueArgs, dp.Timestamp().AsTime().UnixNano())
+			valueArgs = append(valueArgs, dp.Count())
+			valueArgs = append(valueArgs, dp.Sum())
+			valueArgs = append(valueArgs, convertSliceToArraySet(dp.BucketCounts().AsRaw(), logger))
+			valueArgs = append(valueArgs, convertSliceToArraySet(dp.ExplicitBounds().AsRaw(), logger))
+
+			attrs, times, floatValues, intValues, traceIDs, spanIDs := convertExemplars(dp.Exemplars())
+			valueArgs = append(valueArgs, attrs)
+			valueArgs = append(valueArgs, times)
+			valueArgs = append(valueArgs, floatValues)
+			valueArgs = append(valueArgs, intValues)
+			valueArgs = append(valueArgs, traceIDs)
+			valueArgs = append(valueArgs, spanIDs)
+			valueArgs = append(valueArgs, uint32(dp.Flags()))
+			valueArgs = append(valueArgs, dp.Min())
+			valueArgs = append(valueArgs, dp.Max())
+		}
+	}
+
+	if len(valuePlaceholders) == 0 {
+		return nil
+	}
+
+	start := time.Now()
+	_, err := tx.ExecContext(ctx, fmt.Sprintf("%s %s", h.insertSQL, strings.Join(valuePlaceholders, ",")), valueArgs...)
+	if err != nil {
+		return fmt.Errorf("insert histogram metrics fail:%w", err)
+	}
+	duration := time.Since(start)
```

Comment:
might want to capture duration even in case of error

Reply:
done